deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -252,10 +253,12 @@ def __init__(self,
                  mem_efficient_linear=True,
                  ds_config=None,
                  dtype=None):
+        global zero_init_enabled
         self.mem_efficient_linear = mem_efficient_linear
         self.enabled = enabled
         self._set_dtype(ds_config, dtype)
         assert self.dtype in [torch.half, torch.bfloat16, torch.float], f"Invalid data type {self.dtype}, allowed values are [torch.half, torch.bfloat16, torch.float]"
+        zero_init_enabled = True
```

Comment:
Maybe this should only be set in `__enter__`. Just a guess looking at CI errors.